pkg/lib/manifest2po

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/python3
+# This file is part of Cockpit.
+#
+# Copyright (C) 2025 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import argparse
+import collections
+import json
+import pathlib
+import re
+from typing import Any, DefaultDict, Dict, Iterable, List, Set
+
+PO_HEADER = """msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE_VERSION\\n"
+"MIME-Version: 1.0\\n"
+"Content-Type: text/plain; charset=UTF-8\\n"
+"Content-Transfer-Encoding: 8bit\\n"
+"X-Generator: Cockpit manifest2po\\n"
+"""
+
+
+def get_docs_strings(docs: List[Dict[str, str]]) -> Iterable[str]:
+    for doc in docs:
+        assert 'label' in doc, 'doc entry without label'
+        yield doc['label']
+
+
+def get_keyword_strings(keywords: List[Dict[str, List[str]]]) -> Iterable[str]:
+    for keyword in keywords:
+        assert 'matches' in keyword, 'keywords entry without matches'
+        for match in keyword['matches']:
+            yield match
+
+
+def get_menu_strings(menu: Dict[str, Any]) -> Iterable[str]:
```

Comment:
Not crazy about the `Any` here and the no checking of the actual values of the things you're accessing, but I guess there's room for pragmatism here...